src/Bicep.Core/BicepCompiler.cs

```diff
@@ -69,27 +69,30 @@ public async Task<Compilation> CreateCompilation(Uri bicepUri, IReadOnlyWorkspac
         // however we still want to surface as many errors as we can for the module refs that are valid
         // so we will try to restore modules with valid refs and skip everything else
         // (the diagnostics will be collected during compilation)
-        if (await moduleDispatcher.RestoreModules(moduleDispatcher.GetValidModuleReferences(sourceFileGrouping.GetArtifactsToRestore()), forceRestore: forceRestore))
+        var artifactsToRestore = moduleDispatcher.GetValidModuleReferences(
+            sourceFileGrouping.GetExplicitArtifactsToRestore()).Concat(
+                sourceFileGrouping.GetImplicitArtifactsToRestore());
+
+        if (await moduleDispatcher.RestoreModules(artifactsToRestore, forceRestore: forceRestore))
         {
             // modules had to be restored - recompile
-            sourceFileGrouping = SourceFileGroupingBuilder.Rebuild(featureProviderFactory, moduleDispatcher, workspace, sourceFileGrouping);
+            sourceFileGrouping = SourceFileGroupingBuilder.Rebuild(featureProviderFactory, moduleDispatcher, ConfigurationManager, workspace, sourceFileGrouping);
         }
-
         return Create(sourceFileGrouping);
     }
 
-    public async Task<ImmutableDictionary<BicepSourceFile, ImmutableArray<IDiagnostic>>> Restore(Compilation compilation, bool force)
+    public async Task<ImmutableDictionary<BicepSourceFile, ImmutableArray<IDiagnostic>>> Restore(Compilation compilation, bool forceArtifactRestore)
```

Comment:
Why rename the param?

Reply:
Presonally I felt like it makes it easier for the reader to understand what the `bool` indicates without being exposed to the execution context (function name).
So if someone is scanning through the code and sees the value `force` they can tell that its a forcing the restore of artifacts at a glance. 
Would you like me to revert the name back to `force` ?